workflow/rules/sbs_process.smk

```diff
@@ -1,30 +1,19 @@
-from pathlib import Path
 from lib.shared.file_utils import get_filename
-
-PREPROCESS_FP = ROOT_FP / "preprocess"
-SBS_PROCESS_FP = ROOT_FP / "sbs_process"
+from lib.shared.target_utils import output_to_input
 
 
 # Align images from each sequencing round
 rule align:
     conda:
         "../envs/sbs_process.yml"
     input:
-        lambda wildcards: expand(
-            PREPROCESS_FP
-            / "images"
-            / "sbs"
-            / get_filename(
-                {"well": wildcards.well, "tile": wildcards.tile, "cycle": "{cycle}"},
-                "image",
-                "tiff",
-            ),
-            cycle=SBS_CYCLES,
+        lambda wildcards: output_to_input(
+            PREPROCESS_OUTPUTS["convert_sbs"],
```

Comment:
Ok I see. So both the target and actual rules reference these set of outputs?

This eliminates a ton of redundancy.

Reply:
> So both the target and actual rules reference these set of outputs?

Exactly. We only need to specify the outputs and their mapping types then can base all inputs/outputs/targets on this.